src/plasma/plasma_manager.cc

```diff
@@ -1346,10 +1349,21 @@ void process_object_notification(event_loop *loop,
                                  void *context,
                                  int events) {
   PlasmaManagerState *state = (PlasmaManagerState *) context;
-  ObjectInfo object_info;
   /* Read the notification from Plasma. */
-  int error =
-      read_bytes(client_sock, (uint8_t *) &object_info, sizeof(object_info));
+  int64_t size;
```

Comment:
This construct where we call `read_bytes` twice (once to get the size and once to get the data) feels a bit too low-level to be here. It also appears in both the plasma manager and in the local scheduler, which makes me think it should either be in the plasma client library or in `io.cc`. What do you think?